pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -498,7 +506,7 @@ func (r *ReconcileJenkinsBaseConfiguration) checkForPodRecreation(currentJenkins
 		return reason.NewPodRestart(reason.KubernetesSource, messages, verbose...)
 	}
 
-	if userAndPasswordHash != r.Configuration.Jenkins.Status.UserAndPasswordHash {
+	if userAndPasswordHash != r.Configuration.Jenkins.Status.UserAndPasswordHash && r.Configuration.Jenkins.Status.UserAndPasswordHash != "" {
```

Comment:
`val userAndPasswordHashIsDifferent = userAndPasswordHash != r.Configuration.Jenkins.Status.UserAndPasswordHash`

`val userAndPasswordHashStatusNotEmpty = r.Configuration.Jenkins.Status.UserAndPasswordHash != ""`

this way we can avoid adding comments and still make the code easy to understand